src/test/java/uk/gov/hmcts/ccd/data/casedataaccesscontrol/DefaultRoleAssignmentRepositoryTest.java

```diff
@@ -0,0 +1,226 @@
+            applicationParams, securityUtils, restTemplate);
+
+        when(applicationParams.getRoleAssignmentPageSize()).thenReturn(TEST_PAGE_SIZE);
+        when(applicationParams.amQueryRoleAssignmentsURL()).thenReturn(AM_URI);
+
+        when(securityUtils.authorizationHeaders()).thenReturn(new HttpHeaders());
+
+        responseEntity = ResponseEntity.ok().headers(responseHeaders()).body(responseBody());
+        doReturn(responseEntity).when(restTemplate).exchange(eq(AM_URI), eq(POST),
+            any(), eq(RoleAssignmentResponse.class));
+    }
+
+    @DisplayName("should return roleAssignments by user and roles")
+    @Test
+    void shouldReturnRoleAssignmentsByUserAndRoles() {
+        when(applicationParams.isRoleAssignmentPaginationEnabled()).thenReturn(false);
+
+        RoleAssignmentResponse roleAssignmentResponse = defaultRoleAssignmentRepository
+            .findRoleAssignmentsByCasesAndUsers(Collections.singletonList(ATTRIBUTES_CASE_ID),
+                Collections.singletonList(ACTOR_ID));
+
+        assertThat(roleAssignmentResponse.getRoleAssignments().size(), is(1));
+    }
+
+    @DisplayName("should error on 404 when post FindRoleAssignmentsByCasesAndUsers")
+    @Test
+    void shouldErrorOn404WhenPostFindRoleAssignmentsByCasesAndUsers() {
+        when(applicationParams.isRoleAssignmentPaginationEnabled()).thenReturn(false);
+        HttpClientErrorException exception = new HttpClientErrorException(HttpStatus.NOT_FOUND);
+        doThrow(exception).when(restTemplate).exchange(eq(AM_URI), eq(POST), any(), eq(RoleAssignmentResponse.class));
+
+        final ResourceNotFoundException resourceNotFoundException = assertThrows(
+            ResourceNotFoundException.class, () -> defaultRoleAssignmentRepository
+                .findRoleAssignmentsByCasesAndUsers(Collections.singletonList(ATTRIBUTES_CASE_ID),
+                    Collections.singletonList(ACTOR_ID)));
+
+        assertEquals(resourceNotFoundException.getMessage(),
+            String.format(HTTP_404_ERROR_MESSAGE, ACTOR_ID, ATTRIBUTES_CASE_ID));
+    }
+
+    @DisplayName("should error on 500 when post FindRoleAssignmentsByCasesAndUsers")
+    @Test
+    void shouldErrorOn500WhenPostFindRoleAssignmentsByCasesAndUsers() {
+        when(applicationParams.isRoleAssignmentPaginationEnabled()).thenReturn(false);
+        HttpClientErrorException exception = new HttpClientErrorException(HttpStatus.INTERNAL_SERVER_ERROR);
+        doThrow(exception).when(restTemplate).exchange(eq(AM_URI), eq(POST), any(), eq(RoleAssignmentResponse.class));
+
+        final ServiceException serviceException = assertThrows(
+            ServiceException.class, () -> defaultRoleAssignmentRepository
+                .findRoleAssignmentsByCasesAndUsers(Collections.singletonList(ATTRIBUTES_CASE_ID),
+                    Collections.singletonList(ACTOR_ID)));
+
+        assertEquals(serviceException.getMessage(), HTTP_500_ERROR_MESSAGE);
```

Comment:
http-500 from downstream is ideally http-502 to upstream.

Reply:
Existing behaviour. Can be done as part of Kaan's PR or in a new followup tech debt ticket?